src/sidebar/util/test/thread-annotations-test.js

```diff
@@ -0,0 +1,274 @@
+          updated: 20,
+        },
+        {
+          target: targetWithPos(100),
+          updated: 100,
+        },
+        {
+          target: targetWithPos(50),
+          updated: 50,
+        },
+        {
+          target: targetWithPos(20),
+          updated: 10,
+        },
+      ];
+
+      [
+        {
+          order: 'Location',
+          expectedOrder: [0, 3, 2, 1],
+        },
+        {
+          order: 'Oldest',
+          expectedOrder: [3, 0, 2, 1],
+        },
+        {
+          order: 'Newest',
+          expectedOrder: [1, 2, 0, 3],
+        },
+      ].forEach(testCase => {
+        it(`sorts correctly when sorting by ${testCase.order}`, () => {
+          fakeThreadState.selection.sortKey = testCase.order;
+
+          threadAnnotations(fakeThreadState);
+
+          // The sort compare fn passed to `buildThread`
+          const sortCompareFn = fakeBuildThread.args[0][1].sortCompareFn;
+
+          // Sort the test annotations by the sort compare fn that would be
+          // used by `build-thread` and make sure it's as expected
+          const actualOrder = sortBy(annotations, sortCompareFn).map(annot =>
+            annotations.indexOf(annot)
+          );
+          assert.deepEqual(actualOrder, testCase.expectedOrder);
+        });
+      });
+    });
+
+    describe('annotation and thread filtering', () => {
+      context('sidebar route', () => {
+        [
+          uiConstants.TAB_NOTES,
+          uiConstants.TAB_ANNOTATIONS,
+          uiConstants.TAB_ORPHANS,
+        ].forEach(selectedTab => {
+          it(`should filter the thread for the tab '${selectedTab}'`, () => {
+            const annotations = {
+              [uiConstants.TAB_ANNOTATIONS]: {
+                ...annotationFixtures.defaultAnnotation(),
+                $orphan: false,
+              },
+              [uiConstants.TAB_NOTES]: annotationFixtures.oldPageNote(),
+              [uiConstants.TAB_ORPHANS]: {
+                ...annotationFixtures.defaultAnnotation(),
+                $orphan: true,
+              },
+            };
+            const fakeThreads = [
+              {},
+              { annotation: annotations[uiConstants.TAB_ANNOTATIONS] },
+              { annotation: annotations[uiConstants.TAB_NOTES] },
+              { annotation: annotations[uiConstants.TAB_ORPHANS] },
+            ];
+            fakeThreadState.selection.selectedTab = selectedTab;
+
+            threadAnnotations(fakeThreadState);
+
+            const threadFilterFn = fakeBuildThread.args[0][1].threadFilterFn;
+            const filteredThreads = fakeThreads.filter(thread =>
+              threadFilterFn(thread)
+            );
+
+            assert.lengthOf(filteredThreads, 1);
+            assert.equal(
+              filteredThreads[0].annotation,
+              annotations[selectedTab]
+            );
+          });
+        });
+
+        it('should not filter the thread if annotations are filtered', () => {
```

Comment:
This test description confused me. After reading the code I understand that the behavior being tested is "annotations should not be filtered based on the current tab if there is a filter query active".